<a href="https://colab.research.google.com/github/Fernando-cfs/projetomodulo4/blob/main/individualmodulo4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#<font color = red> Projeto Individual Módulo 4

## Sobre o Negócio


A empresa "Soft Sistemas" deseja expandir no mercado global para isso, quer saber como anda o desempenho dos seus funcionários e como pode melhorar para ter maior proveito deles e com isso maior satisfação dos clientes se tornando referência em sua área. Para isso será gerado insights em relação aos seguintes tópicos: Dia, Horas Trabalhadas, Bugs Corrigidos, Tarefas Concluídas e com essas informações teremos insights de como anda o desempenho dos funcionários da 'Soft Sistemas' e como melhorá.




## Importando as bibliotecas
- Pandas
- plotly.express
- plotly.graph_objects

In [ ]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go


## Criação das tabelas



In [ ]:
tabelas = {'Dia': ['Segunda','Terça', 'Quarta','Quinta','Sexta','Sábado', 'Domingo'],
         'Horas Trabalhadas': [6,7,8,6,7,5,4],
              'Bugs Corrigidos': [3,2,1,4,3,2,1],
              'Tarefas Concluídas': [5,4,6,4,5,3,2]}

dftabelas = pd.DataFrame(tabelas)
dftabelas

Dia  Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
0  Segunda                  6                3                   5
1    Terça                  7                2                   4
2   Quarta                  8                1                   6
3   Quinta                  6                4                   4
4    Sexta                  7                3                   5
5   Sábado                  5                2                   3
6  Domingo                  4                1                   2

- A tabela acima mostra as horas, bugs, e tarefas concluídas ao longo da semana, esses dados foi fornecido para analisarmos o desempenho dos funcionários


In [ ]:
tb_bugs_tarefas = pd.DataFrame(dftabelas)
tb_bugs_tarefas['tarefas_bugs'] = tb_bugs_tarefas['Bugs Corrigidos'] + tb_bugs_tarefas['Tarefas Concluídas']
tb_bugs_tarefas

Dia  Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas  \
0  Segunda                  6                3                   5   
1    Terça                  7                2                   4   
2   Quarta                  8                1                   6   
3   Quinta                  6                4                   4   
4    Sexta                  7                3                   5   
5   Sábado                  5                2                   3   
6  Domingo                  4                1                   2   

   tarefas_bugs  
0             8  
1             6  
2             7  
3             8  
4             8  
5             5  
6             3

Foi criado uma nova tabela com a coluna "tarefas_bugs" para somar os "Bugs Corrigidos" e "Tarefas Concluídas" com isso podemos analisar as atividades dos funcionários ao decorre da semana para termos melhores insights


In [ ]:
# Estatística descritivas sobre a tabela
dftabelas.describe()

Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas  tarefas_bugs
count           7.000000         7.000000            7.000000      7.000000
mean            6.142857         2.285714            4.142857      6.428571
std             1.345185         1.112697            1.345185      1.902379
min             4.000000         1.000000            2.000000      3.000000
25%             5.500000         1.500000            3.500000      5.500000
50%             6.000000         2.000000            4.000000      7.000000
75%             7.000000         3.000000            5.000000      8.000000
max             8.000000         4.000000            6.000000      8.000000

Na análise descritiva temos as estatísticas do desempenho dos funcionários da empresa 'Soft Sistemas'
 o mínimo de horas trabalhadas por dia são de 4 horas, o máximo são de 8 horas e cerca de 25% a 75% estão entre 5.5 e 7.0 horas trabalhadas durante a semana,  o número mínimo de bugs e de 1.0 e o máximo de 4.0 por semana e as tarefas concluídas e de no mínimo 2.0 e o máximo 6.0 por semana com isso podemos começar nossa análise e montar os gráficos para uma melhor visualização dos insights gerados

In [ ]:
# correlação
dftabelas.corr(numeric_only=True)

Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas  \
Horas Trabalhadas            1.000000         0.079536            0.907895   
Bugs Corrigidos              0.079536         1.000000            0.190885   
Tarefas Concluídas           0.907895         0.190885            1.000000   
tarefas_bugs                 0.688499         0.719874            0.818755   

                    tarefas_bugs  
Horas Trabalhadas       0.688499  
Bugs Corrigidos         0.719874  
Tarefas Concluídas      0.818755  
tarefas_bugs            1.000000

Na correlação entre bugs e horas trabalhadas não é muito grande, possui uma correlação maior entre horas trabalhadas e tarefas concluídas, o que indica uma maior produtividades nas tarefas concluída e menos bugs durante a semana


## Criação dos cálculos: média diária, total e gráficos


- Total de Horas Trabalhadas
- Média Diária de Horas Trabalhadas

In [ ]:

# Criar DataFrame a partir das tabelas
horas = pd.DataFrame(tabelas)

# Calcular a média e a soma total das horas trabalhadas
media_horas = horas['Horas Trabalhadas'].mean()
total_horas = horas['Horas Trabalhadas'].sum()

# Criar figura com os dados
fig = go.Figure()

# Adicionar as barras para média e total e configuralas
fig.add_trace(go.Bar(x=['Média Diária', 'Total por Semana'],
                     y=[media_horas, total_horas],
                     marker=dict(color=['green', 'red']),
                     text=[f'Média: {media_horas:.2f} horas', f'Total: {total_horas:.2f} horas'],
                     textposition='auto'
                    ))
# Personalizar o layout do gráfico
fig.update_layout(title='Média e Total de Horas Trabalhadas por Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Estatísticas',
                  yaxis_title='Horas',
                  showlegend=False,
                  width=600  # Definindo a largura do gráfico para 600 pixels
                  )
# Mostrar o gráfico interativo
fig.show()


- Nesse gráfico mostra o total e a média de horas trabalhadas por semana, com isso podemos ter uma visão geral sobre como anda a carga horária dos funcionários, se eles estão trabalhando muitas horas por semana podendo baixar o desempenho deles ou estão com pouca carga horária que poderia causar um baixo rendimento nas entregas das tarefas fazendo a produtividade também cair

-------------------------------------------------------------------------------

- Total de Bugs Corrigidos
- Média Diária de Bugs Corrigidos

In [ ]:
# Criar DataFrame a partir das tabelas
bgcorrigidos = pd.DataFrame(tabelas)

# Calcular a média e a soma total dos Bugs Corrigidos
media_bgcorrigidos = horas['Bugs Corrigidos'].mean()
total_bgcorrigidos = horas['Bugs Corrigidos'].sum()

# Criar figura com os dados
fig = go.Figure()

# Adicionar as barras para média e total e configuralas
fig.add_trace(go.Bar(x=['Média Diária', 'Total por Semana'],
                     y=[media_bgcorrigidos, total_bgcorrigidos],
                     marker=dict(color=['green', 'red']),
                     text=[f'Média: {media_bgcorrigidos:.2f} bgcorrigidos', f'Total: {total_bgcorrigidos:.2f} bgcorrigidos'],
                     textposition='auto'
                    ))
# Personalizar o layout do gráfico
fig.update_layout(title='Média e Total de Bugs Corrigidos por Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Estatísticas',
                  yaxis_title='Bugs Corrigidos',
                  showlegend=False,
                  width=600  # Definindo a largura do gráfico para 600 pixels
                  )
# Mostrar o gráfico interativo
fig.show()

- Nesse gráfico podemos observar o total de bugs corrigidos por semana para analisarmos como anda os desempenhos dos funcionários em resolver possíveis erros no sistema e também ter uma média de erros que o sistema está tendo ao decorre da semana, com isso podemos ter uma ideia geral se o serviço está dentro dos padrões de qualidades e ter uma melhor confiança do cliente final

--------------------------------------------------------------------------------

- Total de Tarefas Concluídas
- Média Diária de Tarefas Concluídas

In [ ]:
# Criar DataFrame a partir das tabelas
tconcluidas = pd.DataFrame(tabelas)

# Calcular a média e a soma total dos Tarefas Concluídas
media_tconcluidas = horas['Tarefas Concluídas'].mean()
total_tconcluidas = horas['Tarefas Concluídas'].sum()

# Criar figura com os dados
fig = go.Figure()

# Adicionar as barras para média e total e configuralas
fig.add_trace(go.Bar(x=['Média Diária', 'Total por Semana'],
                     y=[media_tconcluidas, total_tconcluidas],
                     marker=dict(color=['green', 'red']),
                     text=[f'Média: {media_tconcluidas:.2f} tconcluidas', f'Total: {total_tconcluidas:.2f} tconcluidas'],
                     textposition='auto'
                    ))
# Personalizar o layout do gráfico
fig.update_layout(title='Média e Total de Tarefas Concluídas por Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Estatísticas',
                  yaxis_title='Tarefas Concluídas',
                  showlegend=False,
                  width=600  # Definindo a largura do gráfico para 600 pixels
                  )
# Mostrar o gráfico interativo
fig.show()

Essa análise podemos ver a quantidade média e total de tarefas, concluídas durante a semana para entender como anda o desempenho dos funcionários na conclusão das tarefas, esse gráfico vai ajudar a ter uma melhor compreensão das possíveis melhorias ou se está com uma produtividade boa

--------------------------------------------------------------------------------

- Produtividade Diária (Tarefas Concluídas por Hora)

In [ ]:
# Criar DataFrame a partir das tabelas
dftabelas = pd.DataFrame(tabelas)

# Calcular a produtividade diária (tarefas concluídas por hora) para cada dia da semana
dftabelas['Produtividade Diária'] = (dftabelas['Tarefas Concluídas'] / dftabelas['Horas Trabalhadas']).round(2)

# Definir o dia da semana em português
days_of_week = {
    'Segunda': 'Segunda-feira',
    'Terça': 'Terça-feira',
    'Quarta': 'Quarta-feira',
    'Quinta': 'Quinta-feira',
    'Sexta': 'Sexta-feira',
    'Sábado': 'Sábado',
    'Domingo': 'Domingo'
}
dftabelas['Dia da Semana'] = dftabelas['Dia'].map(days_of_week)

# Criar o gráfico de barras com o Plotly
fig = px.bar(dftabelas, x='Dia da Semana', y='Produtividade Diária', text='Produtividade Diária',
             hover_data={'Horas Trabalhadas': True, 'Tarefas Concluídas': True},
             labels={'Produtividade Diária': 'Produtividade Diária (Tarefas Concluídas por Hora)'},
             color_discrete_sequence=px.colors.qualitative.Set2)

# Personalizar o layout do gráfico
fig.update_layout(title='Produtividade Diária ao Longo da Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Dia da Semana',
                  yaxis_title='Produtividade Diária (Tarefas Concluídas por Hora)',
                  xaxis={'categoryorder': 'array', 'categoryarray': list(days_of_week.values())},
                  xaxis_tickangle=-45,
                  barmode='group',
                  bargap=0.2,
                  bargroupgap=0.1)

# Mostrar o gráfico interativo
fig.show()

Nessa análise temos as tarefas concluídas por horas vimos que tem certas irregularidades durante a semana, na segunda teve o maior números de tarefas concluídas por horas e no domingo a menor, segunda teve o total de 6 horas trabalhadas e na quarta-feira 8 horas, mas a quantidade de tarefas entregues por hora ficou abaixo da quantidade na segunda

--------------------------------------------------------------------------------
- Produtividade Diária (Bugs Corrigidos por Hora)

In [ ]:
# Criar DataFrame a partir das tabelas
dftabelas = pd.DataFrame(tabelas)

# Calcular os Bugs Corrigidos por Hora para cada dia da semana
dftabelas['Bugs Corrigidos por Hora'] = (dftabelas['Bugs Corrigidos'] / dftabelas['Horas Trabalhadas']).round(2)

# Definir o dia da semana em português
days_of_week = {
    'Segunda': 'Segunda-feira',
    'Terça': 'Terça-feira',
    'Quarta': 'Quarta-feira',
    'Quinta': 'Quinta-feira',
    'Sexta': 'Sexta-feira',
    'Sábado': 'Sábado',
    'Domingo': 'Domingo'
}
dftabelas['Dia da Semana'] = dftabelas['Dia'].map(days_of_week)

# Criar o gráfico de barras com o Plotly
fig = px.bar(dftabelas, x='Dia da Semana', y='Bugs Corrigidos por Hora', text='Bugs Corrigidos por Hora',
             hover_data={'Horas Trabalhadas': True, 'Bugs Corrigidos': True},
             labels={'Bugs Corrigidos por Hora': 'Bugs Corrigidos (Bugs Corrigidos por Hora)'},
             color_discrete_sequence=px.colors.qualitative.Set2)

# Personalizar o layout do gráfico
fig.update_layout(title='Bugs Corrigidos por Hora ao Longo da Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Dia da Semana',
                  yaxis_title='Bugs Corrigidos (Bugs Corrigidos por Hora)',
                  xaxis={'categoryorder': 'array', 'categoryarray': list(days_of_week.values())},
                  xaxis_tickangle=-45,
                  barmode='group',
                  bargap=0.2,
                  bargroupgap=0.1)

# Mostrar o gráfico interativo
fig.show()

Nesse gráfico podemos ver quantidades de bugs corrigidos por horas durante a semana sendo que na quarta-feira teve 8 horas trabalhadas e menor número de bugs corrigidos e na quinta-feira com 6 horas trabalhadas o maior números de bugs corrigidos podendo ter relação com o quanto de horas trabalhadas na quarta pode refletir em uma grande quantidades de bugs na quinta


--------------------------------------------------------------------------------

- Tarefas_bugs (Tarefas_Bugs por Hora)



In [ ]:
# Criar DataFrame a partir das tabelas
tb_bugs_tarefas = pd.DataFrame(dftabelas)

# Calcular a coluna "tarefas_bugs" somando as colunas "Bugs Corrigidos" e "Tarefas Concluídas"
tb_bugs_tarefas['tarefas_bugs'] = tb_bugs_tarefas['Bugs Corrigidos'] + tb_bugs_tarefas['Tarefas Concluídas']

# Calcular a coluna "tarefas_bugs por Hora" para cada dia da semana
tb_bugs_tarefas['tarefas_bugs por Hora'] = (tb_bugs_tarefas['tarefas_bugs'] / tb_bugs_tarefas['Horas Trabalhadas']).round(2)

# Definir o dia da semana em português
days_of_week = {
    'Segunda': 'Segunda-feira',
    'Terça': 'Terça-feira',
    'Quarta': 'Quarta-feira',
    'Quinta': 'Quinta-feira',
    'Sexta': 'Sexta-feira',
    'Sábado': 'Sábado',
    'Domingo': 'Domingo'
}
tb_bugs_tarefas['Dia da Semana'] = tb_bugs_tarefas['Dia'].map(days_of_week)

# Criar o gráfico de barras com o Plotly
fig = px.bar(tb_bugs_tarefas, x='Dia da Semana', y='tarefas_bugs por Hora', text='tarefas_bugs por Hora',
             hover_data={'Horas Trabalhadas': True, 'tarefas_bugs': True},
             labels={'tarefas_bugs por Hora': 'tarefas_bugs (tarefas_bugs por Hora)'},
             color_discrete_sequence=px.colors.qualitative.Set2)

# Personalizar o layout do gráfico
fig.update_layout(title='tarefas_bugs por Hora ao Longo da Semana',
                  title_x=0.5,  # Alinhar o título no centro do gráfico
                  xaxis_title='Dia da Semana',
                  yaxis_title='tarefas_bugs (tarefas_bugs por Hora)',
                  xaxis={'categoryorder': 'array', 'categoryarray': list(days_of_week.values())},
                  xaxis_tickangle=-45,
                  barmode='group',
                  bargap=0.2,
                  bargroupgap=0.1)

# Mostrar o gráfico interativo
fig.show()


- Nessa análise temos a soma das tabelas de "Bugs Corrigidos" mais as "Tarefas Concluídas" criando a tabela "tarefas_bugs" para vermos o desempenho da equipe durante a semana e tiramos uma melhor conclusão de todas as atividades que estão sendo feitas por hora, na "segunda-feira" e "quinta-feira" foram 6 horas, e 8 atividades concluídas sendo que "segunda-feira" foram 5 "Tarefas Concluídas" e 3 "Bugs Corrigidos" na "quinta-feira" foram 4 "Tarefas Concluídas" e 4 "Bugs Corrigidos" com isso vimos que a equipe leva o mesmo tempo para realizar tarefas ou corrigir bugs

## Relatório
Ao analisar os dados da empresa "Soft Sistemas" podemos tira conclusões para identificar ponto de melhoria para os funcionário e a empresa buscando assim o melhor serviço para o cliente final

- Houve uma média diária de 6.14 horas trabalhadas com uma média de 4.14 de tarefas concluídas sendo a média de 2.29 bugs corrigidos; com isso podemos ter uma média do desempenho dos funcionários e dos trabalhos realizados no decorre da semana
- O total de horas trabalhadas por semana é de 43 horas, o total de tarefas concluídas é de 29 tarefas, e o total de bugs corrigidos é de 16 bugs; temos uma ideia geral do tempo de serviço e atividades concluídas
- Produtividade Diária (Tarefas Concluídas por Hora) na segunda-feira houve o maior desempenho de tarefas concluídas por horas e no domingo o menor;
- Conforme a análise realizada da quantidade de bugs corrigidos por hora: na quarta-feira ocorreu o menor número de bugs corrigidos e na quinta-feira houve o maior número de bugs corrigidos em relação às horas trabalhadas.
- Analisando os "Bugs Corrigidos" mais as "Tarefas Concluídas" por horas trabalhadas, podemos ver que tem dias que a quantidades de atividades e igual ou superior às horas trabalhadas, com isso alguns dias sobra tempo para realizar as atividades podendo padronizar os horários para 6 horas que tiveram o melhor desempenho concluindo muitas tarefas e corrigindo bugs com o total de 8 atividades com isso teremos uma melhor gestão da equipe e melhor desempenho dos funcionários